### PR TITLE
Fix V1 setters

### PR DESCRIPTION
**Pull request by phanimarupaka**
@mortent 

This is the PR to fix the broken support for setters v1.

### CODE REVIEW

File: kyaml/fieldmeta/fieldmeta.go [hunk 1]
```diff
@@ -254,6 +274,8 @@ const (
 // shortHandRef is the shorthand reference to setters and substitutions
 var shortHandRef = "$openapi"
 
+var SetterVersion = "v2"
```
Comment (mortent) [inline kyaml/fieldmeta/fieldmeta.go:277]: Can we avoid this? This seems like something that should be a function parameter that we have moved to a package level variable. This makes it much harder to use the library in other situations.

Comment (phanimarupaka) [reply kyaml/fieldmeta/fieldmeta.go]: I felt it's okay for it to be package level variable as the setter version is constant per command invocation. But avoided it and updated the PR. PTAL.

File: kyaml/fieldmeta/fieldmeta.go [hunk 2]
```diff
@@ -145,16 +145,14 @@ func isExtensionEmpty(x XKustomize) bool {
 
 // Write writes the FieldMeta to a node
 func (fm *FieldMeta) Write(n *yaml.RNode) error {
-	if !isExtensionEmpty(fm.Extensions) {
-		fm.Schema.VendorExtensible.AddExtension("x-kustomize", fm.Extensions)
-	} else {
-		delete(fm.Schema.VendorExtensible.Extensions, "x-kustomize")
+	if len(fm.Extensions.PartialFieldSetters) > 0 || fm.Extensions.FieldSetter != nil {
```
Comment (monopole) [inline kyaml/fieldmeta/fieldmeta.go:148]: nit: please factor to a boolean method named to explain what this is testing

File: kyaml/fieldmeta/fieldmeta.go [hunk 3]
```diff
@@ -174,6 +172,22 @@ func (fm *FieldMeta) Write(n *yaml.RNode) error {
 	return nil
 }
 
+// WriteV1Setters is the v1 setters way of writing setter definitions
+// TODO: pmarupaka - remove this method after migration
+func (fm *FieldMeta) WriteV1Setters(n *yaml.RNode) error {
+	if !isExtensionEmpty(fm.Extensions) {
```
Comment (monopole) [inline kyaml/fieldmeta/fieldmeta.go:178]: nit: true clause first - easier to read if not notted :-)

Comment (phanimarupaka) [reply kyaml/fieldmeta/fieldmeta.go]: I missed that. Thanks for pointing. Addressed.

**Comment by monopole** [comment]
well that got simpler

/lgtm

**Pull request merged**